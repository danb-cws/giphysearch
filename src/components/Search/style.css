.SearchInput {
  font-size: 3em;
}
.SearchInput::-ms-clear {
  display: none;
}