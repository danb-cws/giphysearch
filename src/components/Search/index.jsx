import React, { Component } from "react";
import * as config from "../../config";
import debounce from "debounce";
import "./style.css";

class Search extends Component {
  state = { searchTerm: config.default_term };

  onInputChange = searchTerm => {
    this.setState({ searchTerm }); // dont want to debounce this as the controlled input gets laggy
    this.debouncedOnInputChange(searchTerm);
  };

  debouncedOnInputChange = debounce(searchTerm => {
    this.props.onSearchTermChange(searchTerm);
  }, 200);

  componentDidMount() {
    this.onInputChange(this.state.searchTerm); // fire with initial default search term
  }

  render() {
    return (
      <label className="Search">
        <span className="Search__label">Search term</span>
        <input
          type="text"
          autoFocus
          className="Search__input"
          value={this.state.searchTerm}
          onChange={e => this.onInputChange(e.target.value)}
        />
      </label>
    );
  }
}

export default Search;
