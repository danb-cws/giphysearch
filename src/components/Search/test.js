import React from "react";
import { mount } from "enzyme";
import Search from "./";

const el = mount(
  <Search />
);

describe("Search tests", () => {
  it("Renders the Search component", () => {
    expect(el.exists()).toBe(true);
  });

  it("Renders and matches snapshot", () => {
    expect(el).toMatchSnapshot();
  });

  it("Responds to onChange on the input", () => {
    el.find('input').simulate("change", {
      target: { value: "hello" }
    });
    expect(el.state("searchTerm")).toEqual("hello");
  });
});
