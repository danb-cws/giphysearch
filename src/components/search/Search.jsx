import React, { Component } from "react";
import * as config from "../../config";
import "./Search.css";

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = { searchTerm: config.default_term };
  }

  render() {
    return (
      <input
        autoFocus
        className="SearchInput"
        value={this.state.searchTerm}
        onChange={e => this.onInputChange(e.target.value)}
      />
    );
  }

  onInputChange = searchTerm => {
    this.setState({ searchTerm });
    this.props.onSearchTermChange(searchTerm);
  };
}

export default Search;
