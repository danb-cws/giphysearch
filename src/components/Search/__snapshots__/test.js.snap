// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search tests Renders and matches snapshot 1`] = `
<Search>
  <label
    className="Search"
  >
    <span
      className="Search__label"
    >
      Search term
    </span>
    <input
      autoFocus={true}
      className="Search__input"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
</Search>
`;
